cypress/support/cypress-commands/assertion.js

```diff
@@ -18,120 +18,68 @@
-    const fetchErrorValidationObjectForExceptionmethod = (method, param) => {
-      return new Promise((resolve, reject) => {
-        const exceptionMethods = UTILS.getEnvVariable(CONSTANTS.EXCEPTION_METHODS);
-        let exceptionType, methodInExceptionList, errorType;
-        // Looping through the different types of exceptions lists.
-        for (const [type, list] of Object.entries(exceptionMethods)) {
-          // Looking for the method and params in each list, if matched returning that exception method.
-          methodInExceptionList = list.find((obj) => {
-            if (obj.method && obj.method.toLowerCase() === method.toLowerCase()) {
-              if (obj.hasOwnProperty(CONSTANTS.PARAM)) {
-                return _.isEqual(obj.param, param);
-              } else {
-                return true;
-              }
-            }
-            return false;
-          });
+  (errorResponse, errorSchemaObject, apiObject, isExceptionMethod) => {
+    errorSchemaObject = isExceptionMethod
+      ? fetchErrorValidationObjectForExceptionMethod()
+      : errorSchemaObject;
 
-          // If method is prsent in the list, exiting the loop.
-          if (methodInExceptionList) {
-            exceptionType = type;
-            break;
-          }
-        }
-
-        if (methodInExceptionList) {
-          switch (exceptionType) {
-            case CONSTANTS.NOT_AVAILABLE_METHODS:
-              errorType = CONSTANTS.NOT_AVAILABLE;
-              break;
-            case CONSTANTS.NOT_PERMITTED_METHODS:
-              errorType = CONSTANTS.NOT_PERMITTED;
-              break;
-            default:
-              errorType = CONSTANTS.NOT_SUPPORTED;
-          }
-        }
-
-        // Extracting the error content validation object from the environment variable.
-        if (
-          UTILS.getEnvVariable(CONSTANTS.ERROR_CONTENT_VALIDATIONJSON, false) &&
-          UTILS.getEnvVariable(CONSTANTS.ERROR_CONTENT_VALIDATIONJSON)[errorType]
-        ) {
-          const contentObject = UTILS.getEnvVariable(CONSTANTS.ERROR_CONTENT_VALIDATIONJSON)[
-            errorType
-          ];
-          resolve(contentObject);
-        } else {
-          reject(`Unable to find Error content validation object for ${errorType}`);
-        }
-      });
-    };
-
-    if (expectedContent === CONSTANTS.EXCEPTION_ERROR_OBJECT) {
-      cy.wrap(fetchErrorValidationObjectForExceptionmethod(method, param)).then((errorObject) => {
-        validateErrorObjects(errorObject);
-      });
+    if (errorSchemaObject.type === 'schemaOnly') {
+      return;
+    } else if (errorSchemaObject.type === CONSTANTS.CUSTOM) {
+      const additionalParams = { apiObject: apiObject, errorSchemaObject: errorSchemaObject };
+      cy.customValidation(errorSchemaObject, additionalParams);
     } else {
-      validateErrorObjects(expectedContent);
+      validateErrorObjects(errorSchemaObject);
     }
 
```

Comment:
why do we have errorSchemaObject passed in line 47 ? and in line 51, this becomes errorContentObject

Reply:
addressed the comment bu changing to expectedContet